## STAT 207 Lab 4: Random Variables - [25 total points] - Solutions

Due: Wednesday, March 10, 11:59pm CST



## Imports

In [1]:
import pandas as pd
import numpy as np

## <u>Question 1</u>: Bernoulli Random Variables

In the previous year the acceptance rate at UIUC was 62\%. Suppose we conduct an experiment where we randomly select one of the UIUC applicants from the previous year and observe whether they were accepted into UIUC. We're interested to know if the applicant was accepted.


The Bernoulli random variable object in Python operates has similar functions to the other random variables that we have discussed in this class. The documentation for the functions that correspond to the bernoulli random variable are shown here.
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.bernoulli.html

### 1.1 [2 pts] Define a Bernoulli random variable that models this experiment.
* Make sure you define any parameters that pertain to this random variable as well as define the success level.
(Just put it into words).

Let X be a Bernoulli random variable, where the probability that a student was accepted is 0.62, and the probability that they were not accepted is 0.38.

### 1.2 [2 pts]  Calculate the mean, variance, and standard deviation of this random variable.

E(X) = 0*(.38) + 1(.62) = .62

V(X) = (0-mu)^2 * P(x=0) + (1-mu)^2 * P(x=1)
     = (0 - p)^2 * (1-p) + (1-p)^2 * p
     = p^2(1-p) + (1-p)^2 * p
     = p(1-p)
     = .62(.38)
     = .2356

SD(X) = sqrt(V(X)
      = 0.485386444

### 1.3 [2 pts]  Generate 1000 random values for this random variable and put them into a pandas dataframe.

In [9]:
from scipy.stats import bernoulli
sample = bernoulli.rvs(p=0.62,size=1000)
sample = pd.DataFrame(sample)
sample

0
0    1
1    1
2    0
3    1
4    1
..  ..
995  1
996  0
997  0
998  0
999  0

[1000 rows x 1 columns]

### 1.4 [2 pts]  Calculate the mean, variance, and standard deviation of these randomly generated random variable values. 

In [10]:
sample.mean()

0    0.616
dtype: float64

In [13]:
sample.std() ** 2

0    0.236781
dtype: float64

In [12]:
sample.std()

0    0.486601
dtype: float64

### 1.5 [2 pts]  How close are your mean, variance, and standard deviation from 1.4 to the mean, variance, and standard deviation that you calculated in 1.2?

The difference in my two calculations are: |.62 - .616| = .004

The difference in my two calculations are: |.2356 - .236781| = .001181

The difference in my two calculations are: |.485386444 - .486601| = 0.001215

My calculations in 1.4 are very close to the calculations in 1.2.

## <u>Question 2</u>: More Random Variables

Suppose a gambler has a strategy to keep playing a slot matchine until he wins a round. After he **wins** a round, he stops playing for the day. The probability of **losing** any given slot machine round is 0.99.

### 2.1 [2 pts]  On average, how many rounds of the slot machine will the gambler play before stopping for the day? 



Let X be a Bernoulli random variable, where the probability that the gambler wins a round on the slot machine is 0.01, and the probability that he doesn not win is 0.99.

On average, it will take the gambler 99 rounds of losing to stop playing for the day.

### 2.2 [2 pts]  Suppose that it costs $\$1$ to play each round of the slot machine. The gambler always brings $\$50$ with him to play the slot machine each day. If he runs out of this $\$50$ (ie. he plays the game more than 50 times), then he goes to borrow money from his friend and keeps playing the slot machine until he wins. What percent of days does he need to borrow money from this friend?

99% of days

## <u>Question 3</u>: Normal Random Variables

Suppose the distribution of SAT scores for the seniors at a local high school (just math and verbal) has an average of 1000, a standard deviation of 100, and follows a normal distribution. 

### 3.1 [2 pts]  What is the probability that randomly selected senior from the high school scored more than 1250?

Using normal cdf....

In [22]:
from scipy.stats import norm
x = norm.cdf(1250, loc=1000, scale = 100)
print (1-x)

0.006209665325776159


### 3.2 [2 pts]  What percent of seniors at the school made score between a 900 and 950?

Using normal cdf...

In [23]:
norm.cdf(950, loc = 1000, scale = 100) - norm.cdf(900, loc = 1000, scale = 100)

0.1498822847945298

### 3.3 [3 pts]  Suppose that all seniors at the school who score higher than a 1200 get a scholarship from the county. Suppose that we randomly select a student that we KNOW got the scholarship. GIVEN that we know that the student has the scholarship, what is the probability that this student's SAT score was less than 1300?

#### Hint: You might need to consider some other probability rules that we've encountered in this class.

The probability that a given student scored higher than 1200 is 1 - norm.cdf(1200, loc = 1000, scale = 100)

In [30]:
prob_of_1200 = 1 - norm.cdf(1200, loc = 1000, scale = 100)
prob_of_1300 = 1 - norm.cdf(1300, loc = 1000, scale = 100)

0.02275013194817921

In [33]:
print(1 - (prob_of_1300/prob_of_1200))

0.9406641669285728


The probability that a student with a scholarship scored below a 1300 is about 94%

## <u>Question 4</u>: Exploring Properties of the Sample Mean Random Variable

Let's revisit our distribution of SAT scores for the seniors at a local high school (just math and verbal) that had an average of 1000, a standard deviation of 100, and followed a normal distribution. 

Let's define **$X_1$=SAT score of a randomly selected high school senior**

### 4.1 [2 pts] What is $E[X_1]$,  $V[X_1]$, and $SD[X_1]$? (looking for actual numbers).

E(X) = mu = 1000
V(X) = std^2 = 10000
SD(X) = std = 100

### <u>Tutorial</u>: Functions of Random Variables

Remember from Unit 7, that we can create new random variables by adding and multiplying random variable by other random variables or other numbers.

For instance, if we have the two following random variables,

**$X_1$=SAT score of a randomly selected high school senior**

**$X_2$=SAT score of a randomly selected high school senior**

then $\bar{X}=\frac{X_1+X_2}{2}$ is also a random variable.

### <u>Tutorial</u>: Properties of the $E[ ]$ function and the $V[]$ function.

* Recall that if we are dealing with a random variable Y that is associated with an experiment, then 
    - if we were to repeat this experiment many, many times, calculate the value of Y for the outcome of each of these experiments, and then take the **average** of these Y values, we would expect this average to be very close to $E[Y]$.
    - if we were to repeat this experiment many, many times, calculate the value of Y for the outcome of each of these experiments, and then take the **variance** of these Y values, we would expect this average to be very close to $V[Y]$.
    

We can think of these $E[ ]$ and $V[]$ as **functions** that also happen to have certain properties.


#### <u>Property of the $E[ ]$ function</u>
For instance, if X and Y ar|e random variables and a and b are coefficients, then
$E[aX+bY] = a\cdot E[X]+b\cdot E[Y]$.

For instance:

$E[7X_1+4X_2] = 7\cdot E[X_1]+4\cdot E[X_2]$.

### 4.2  [2 pts] Calculating $E[\bar{X}]$.
Let's define the following four random variables:

* **$X_1$=SAT score of a randomly selected high school senior**

* **$X_2$=SAT score of a randomly selected high school senior**

* **$X_3$=SAT score of a randomly selected high school senior**

* **$X_4$=SAT score of a randomly selected high school senior**.

Specifically, we say that $X_i \sim N(\mu = 1000, \sigma=100)$ for i=1, i=2, i=3, and i=4.

Finally, let's define one more random variable as:
* **$\bar{X}=\frac{X_1+X_2+X_3+X_4}{4}$**


#### Use the properties from the tutorials above to calculate $E[\bar{X}]$.

This is a proof, so show your work!

Xbar = (X1 + X2 + X3 + X4)/4
     = X1/4 + X2/4 + X3/4 + X4/4
     = 1000/4 + 1000/4 + 1000/4 + 1000/4
     = 250 + 250 + 250 + 250
     = 1000
     